### PR TITLE
Fix SLC generation for all boards

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem / Feature
* Ble Refactor highlighted the fact that all the generated files were generated with the same GSDK version

#### Change overview
* Regenerate all boards with gsdk 4.1.1

#### Testing
* CI to verify everything builds
* Manual tests with BRD4164a and BRD4186c

### CODE REVIEW

File: matter/efr32/efr32mg12/BRD4304A/config/sl_bluetooth_config.h [hunk 1]
```diff
@@ -103,11 +101,11 @@
 #endif // SL_COMPONENT_CATALOG_PRESENT
 
 #ifdef SL_CATALOG_KERNEL_PRESENT
-// void sli_bt_rtos_ll_callback();
-// void sli_bt_rtos_stack_callback();
+void sli_bt_rtos_ll_callback();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg12/BRD4304A/config/sl_bluetooth_config.h:104]: indentation

File: matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.c [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
-void sl_platform_init(void) {
+void sl_platform_init(void)
+{
   CHIP_Init();
   sl_device_init_nvic();
   sl_board_preinit();
   sl_device_init_dcdc();
   sl_hfxo_manager_init_hardware();
   sl_device_init_hfxo();
+  sl_device_init_lfrco();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.c:40]: Why this one ads lfrco and why it has both lfrco and lfxo

File: matter/efr32/efr32mg24/BRD4186A/autogen/sl_device_init_clocks.c [hunk 3]
```diff
@@ -33,12 +33,12 @@
 
 sl_status_t sl_device_init_clocks(void)
 {
-    CMU_ClockSelectSet(cmuClock_SYSCLK, cmuSelect_HFXO);
+    CMU_ClockSelectSet(cmuClock_SYSCLK, cmuSelect_HFRCODPLL);
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186A/autogen/sl_device_init_clocks.c:36]: Did the 4186A really swapped from HFXO to HFRCODPLL ?

File: matter/efr32/efr32mg24/BRD4186A/autogen/sl_event_handler.c [hunk 4]
```diff
@@ -34,7 +38,9 @@ void sl_platform_init(void)
     sl_device_init_dcdc();
     sl_hfxo_manager_init_hardware();
     sl_device_init_hfxo();
+    sl_device_init_lfrco();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186A/autogen/sl_event_handler.c:41]: LFRCO and LFXO again seems odd

File: matter/efr32/efr32mg24/BRD4186A/config/sl_device_init_hfxo_config.h [hunk 5]
```diff
@@ -16,7 +46,7 @@
 
 // <o SL_DEVICE_INIT_HFXO_CTUNE> CTUNE <0-255>
 // <i> Default: 140
-#define SL_DEVICE_INIT_HFXO_CTUNE          140
+#define SL_DEVICE_INIT_HFXO_CTUNE          87
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186A/config/sl_device_init_hfxo_config.h:49]: Why did CTUNE change to non default value

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c [hunk 6]
```diff
@@ -36,6 +39,7 @@ void sl_platform_init(void)
   sl_device_init_dcdc();
   sl_hfxo_manager_init_hardware();
   sl_device_init_hfxo();
+  sl_device_init_lfrco();
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c:42]: LFRCO and LFXO

File: matter/efr32/efr32mg24/BRD4187A/autogen/sl_component_catalog.h [hunk 7]
```diff
@@ -21,33 +22,41 @@
...
 #define SL_CATALOG_GATT_CONFIGURATION_PRESENT
-
 #ifdef DISPLAY_ENABLED
 #define SL_CATALOG_MEMLCD_USART_PRESENT
+#define SL_CATALOG_DMD_MEMLCD_PRESENT
 #endif
+#define SL_CATALOG_NVM3_PRESENT
+#define SL_CATALOG_PSA_CRYPTO_PRESENT
 #define SL_CATALOG_RAIL_LIB_PRESENT
+#define SL_CATALOG_RAIL_UTIL_PTI_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187A/autogen/sl_component_catalog.h:38]: #define SL_CATALOG_SE_MANAGER_PRESENT still there

File: matter/efr32/efr32mg24/BRD4187A/autogen/sl_device_init_clocks.c [hunk 8]
```diff
@@ -33,12 +33,12 @@
 
 sl_status_t sl_device_init_clocks(void)
 {
-    CMU_ClockSelectSet(cmuClock_SYSCLK, cmuSelect_HFXO);
+    CMU_ClockSelectSet(cmuClock_SYSCLK, cmuSelect_HFRCODPLL);
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187A/autogen/sl_device_init_clocks.c:36]: Like brd4186A, the swap to HFRCODPLL seems odd

File: matter/efr32/efr32mg24/BRD4187A/config/sl_device_init_hfxo_config.h [hunk 9]
```diff
@@ -16,7 +46,7 @@
 
 // <o SL_DEVICE_INIT_HFXO_CTUNE> CTUNE <0-255>
 // <i> Default: 140
-#define SL_DEVICE_INIT_HFXO_CTUNE          140
+#define SL_DEVICE_INIT_HFXO_CTUNE          95
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187A/config/sl_device_init_hfxo_config.h:49]: Non default CTUNE here too

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h [hunk 10]
```diff
@@ -21,16 +22,20 @@
...
 #define SL_CATALOG_GATT_CONFIGURATION_PRESENT
-
 #ifdef DISPLAY_ENABLED
 #define SL_CATALOG_MEMLCD_USART_PRESENT
+#define SL_CATALOG_DMD_MEMLCD_PRESENT
 #endif
+#define SL_CATALOG_NVM3_PRESENT
+#define SL_CATALOG_PSA_CRYPTO_PRESENT
 #define SL_CATALOG_RAIL_LIB_PRESENT
+#define SL_CATALOG_RAIL_UTIL_PTI_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h:38]: #define SL_CATALOG_SE_MANAGER_PRESENT is still there

File: matter/efr32/mgm24/BRD4317A/autogen/sl_component_catalog.h [hunk 11]
```diff
@@ -28,8 +31,12 @@
 
 #ifdef DISPLAY_ENABLED
 #define SL_CATALOG_MEMLCD_USART_PRESENT
+#define SL_CATALOG_DMD_MEMLCD_PRESENT
 #endif
+#define SL_CATALOG_NVM3_PRESENT
+#define SL_CATALOG_PSA_CRYPTO_PRESENT
 #define SL_CATALOG_RAIL_LIB_PRESENT
+#define SL_CATALOG_RAIL_UTIL_PTI_PRESENT
 #define SL_CATALOG_SE_MANAGER_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/mgm24/BRD4317A/autogen/sl_component_catalog.h:40]: #define SL_CATALOG_SE_MANAGER_PRESENT still there

File: matter/efr32/mgm24/BRD4319A/autogen/sl_component_catalog.h [hunk 12]
```diff
@@ -30,8 +33,10 @@
     This devices does not support LCD
     Removing all the LCD components
 */
-
+#define SL_CATALOG_NVM3_PRESENT
+#define SL_CATALOG_PSA_CRYPTO_PRESENT
 #define SL_CATALOG_RAIL_LIB_PRESENT
+#define SL_CATALOG_RAIL_UTIL_PTI_PRESENT
 #define SL_CATALOG_SE_MANAGER_PRESENT
```
Comment (jmartinez-silabs) [inline matter/efr32/mgm24/BRD4319A/autogen/sl_component_catalog.h:40]: #define SL_CATALOG_SE_MANAGER_PRESENT still there

File: matter/efr32/efr32mg12/BRD4162A/autogen/sl_component_catalog.h [hunk 13]
```diff
@@ -19,26 +21,35 @@
...
 #endif
-#if CHIP_DEVICE_CONFIG_ENABLE_SED
-#define SL_CATALOG_POWER_MANAGER_PRESENT
-#define SL_CATALOG_SLEEPTIMER_PRESENT
-#endif // CHIP_DEVICE_CONFIG_ENABLE_SED
+
 #define SL_CATALOG_NVM3_PRESENT
+#define SL_CATALOG_PSA_CRYPTO_PRESENT
 #define SL_CATALOG_RAIL_LIB_PRESENT
+#define SL_CATALOG_SE_MANAGER_PRESENT
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4162A/autogen/sl_component_catalog.h:38]: done

File: matter/efr32/efr32mg12/BRD4164A/config/sl_bluetooth_config.h [hunk 14]
```diff
@@ -103,11 +101,11 @@
 #endif // SL_COMPONENT_CATALOG_PRESENT
 
 #ifdef SL_CATALOG_KERNEL_PRESENT
-// void sli_bt_rtos_ll_callback();
-// void sli_bt_rtos_stack_callback();
+void sli_bt_rtos_ll_callback();
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4164A/config/sl_bluetooth_config.h:104]: done

File: matter/efr32/efr32mg12/BRD4164A/config/sl_memory_config.h [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+
+// <<< Use Configuration Wizard in Context Menu >>>
+// <h> Memory configuration
+
+// <o SL_STACK_SIZE> Stack size for the application.
+// <i> Default: 4096
+// <i> The stack size configured here will be used by the stack that the
+// <i> application uses when coming out of a reset.
+#ifndef SL_STACK_SIZE
+  #define SL_STACK_SIZE  4096
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4164A/config/sl_memory_config.h:12]: done

File: matter/efr32/efr32mg12/BRD4164A/config/sl_memory_config.h [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+#endif
+
+// <o SL_HEAP_SIZE> Minimum heap size for the application.
+// <i> Default: 2048
+// <i> Note that this value will configure the c heap which is normally used by
+// <i> malloc() and free() from the c library. The value defines a minimum heap
+// <i> size that is guaranteed to be available. The available heap may be larger
+// <i> to make use of any memory that would otherwise remain unused.
+#ifndef SL_HEAP_SIZE
+  #define SL_HEAP_SIZE   2048
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4164A/config/sl_memory_config.h:22]: done

File: matter/efr32/efr32mg12/BRD4164A/config/sl_uartdrv_usart_vcom_config.h [hunk 17]
```diff
@@ -55,7 +55,7 @@
 // <usartOVS6=> 6x oversampling
 // <usartOVS4=> 4x oversampling
 // <i> Default: usartOVS16
-#define SL_UARTDRV_USART_VCOM_OVERSAMPLING      usartOVS16
+#define SL_UARTDRV_USART_VCOM_OVERSAMPLING      usartOVS4
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4164A/config/sl_uartdrv_usart_vcom_config.h:58]: slc does not agree

File: matter/efr32/efr32mg12/BRD4166A/autogen/sl_component_catalog.h [hunk 18]
```diff
@@ -19,22 +21,30 @@
 #define SL_CATALOG_BLUETOOTH_FEATURE_SYSTEM_PRESENT
 #define SL_CATALOG_BLUETOOTH_PRESENT
 #define SL_CATALOG_BLUETOOTH_FEATURE_MULTIPROTOCOL_PRESENT
+#define SL_CATALOG_GECKO_BOOTLOADER_INTERFACE_PRESENT
 #define SL_CATALOG_DEVICE_INIT_NVIC_PRESENT
+#define SL_CATALOG_DMD_MEMLCD_PRESENT
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply matter/efr32/efr32mg12/BRD4166A/autogen/sl_component_catalog.h:26]: done

File: matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.c (hunk 2 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.c:40]: slc says its necessary

File: matter/efr32/efr32mg24/BRD4186A/autogen/sl_event_handler.c (hunk 4 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4186A/autogen/sl_event_handler.c:41]: slc generation

File: matter/efr32/efr32mg24/BRD4186A/config/sl_device_init_hfxo_config.h (hunk 5 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4186A/config/sl_device_init_hfxo_config.h:49]: slc generation, my guess is the original value was for rev0?

File: matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c (hunk 6 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4186C/autogen/sl_event_handler.c:42]: idem

File: matter/efr32/efr32mg24/BRD4187A/autogen/sl_component_catalog.h (hunk 7 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4187A/autogen/sl_component_catalog.h:38]: done

File: matter/efr32/efr32mg24/BRD4187A/autogen/sl_device_init_clocks.c (hunk 8 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4187A/autogen/sl_device_init_clocks.c:36]: when adding the dpll componenent, clock value is set to that

File: matter/efr32/efr32mg24/BRD4187A/config/sl_device_init_hfxo_config.h (hunk 9 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4187A/config/sl_device_init_hfxo_config.h:49]: idem slc

File: matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h (hunk 10 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4187C/autogen/sl_component_catalog.h:38]: done

File: matter/efr32/mgm24/BRD4317A/autogen/sl_component_catalog.h (hunk 11 above)
Comment (mkardous-silabs) [reply matter/efr32/mgm24/BRD4317A/autogen/sl_component_catalog.h:40]: done

File: matter/efr32/mgm24/BRD4319A/autogen/sl_component_catalog.h (hunk 12 above)
Comment (mkardous-silabs) [reply matter/efr32/mgm24/BRD4319A/autogen/sl_component_catalog.h:40]: done

File: matter/efr32/efr32mg24/BRD4186A/autogen/sl_device_init_clocks.c (hunk 3 above)
Comment (mkardous-silabs) [reply matter/efr32/efr32mg24/BRD4186A/autogen/sl_device_init_clocks.c:36]: adding the DPLL componenent changes the clock selection

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**